src/main/main.go

```diff
@@ -0,0 +1,373 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	v13 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+	"os"
+	"skupper_ocp_smoke/pkg/skupper_ocp_smoke"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func setup(ctx context.Context, pubCli *skupper_ocp_smoke.Client, privCli *skupper_ocp_smoke.Client) error {
+
+	cliList := []*skupper_ocp_smoke.Client{
+		pubCli,
+		privCli,
+	}
+
+	step := "setup"
+	log.Printf("%s : Starting Setup", step)
+
+	//
+	// Delete the namespace, if set by EnvVar
+	//
+	if strings.ToUpper(os.Getenv("CLEANBEFORE")) == "TRUE" {
+		skupper_ocp_smoke.PrintIfDebug(step, ": Removing namespaces")
+		for _, cli := range cliList {
+			skupper_ocp_smoke.PrintIfDebug(step, ": Remove namespace", cli.Namespace)
+			if err := cli.DeleteNamespace(ctx, step); err != nil {
+				return fmt.Errorf("%s : %s", step, err.Error())
+			}
+			if err := cli.WaitForNSDeletion(ctx, cli.WaitLimit(), step); err != nil {
+				return fmt.Errorf("%s : %s", step, err.Error())
+			}
+		}
+	}
+
+	// Create the namespace
+	log.Printf("%s : Creating namespaces", step)
+	for _, cli := range cliList {
+		skupper_ocp_smoke.PrintIfDebug(step, ": Creating namespace", cli.Namespace)
+		if err := cli.CreateNamespace(ctx, step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+	}
+
+	// Check if the Skupper operator is available
+	// By default it checks for the operator "skupper-operator" in namespace "openshift-marketplace",
+	// But that can be configured using ENV_VARS
+	if err := pubCli.IsSkupperOperatorAvailable(step, pubCli.OperatorName(), pubCli.OperatorNameSpace(), pubCli.OperatorCatalog()); err != nil {
+		return fmt.Errorf("%s : %s", step, err.Error())
+	}
+
+	// Create the Catalog Groups
+	log.Printf("%s : Create OperatorGroups", step)
+	for _, cli := range cliList {
+		// Create the operatorgroup
+		if err := cli.CreateOperatorGroup(step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
```

Comment:
It is failing here when I am using CLEANBEFORE=TRUE. I believe it should simply
ignore the "already exists" error in such case.

Reply:
Fixed for all the deployment steps, it will fail only if the Job definition already exists